### PR TITLE
RavenDB-14223 Various rollup and retention fixes

### PR DESCRIPTION
**Pull request by karmeli87**

### CODE REVIEW

File: src/Raven.Client/Documents/Operations/TimeSeries/TimeSeriesConfiguration.cs [hunk 1]
```diff
@@ -8,11 +9,16 @@ public class TimeSeriesConfiguration : IDynamicJson
         internal const char TimeSeriesRollupSeparator = '@';
         public Dictionary<string, TimeSeriesCollectionConfiguration> Collections { get; set; }
 
+        public TimeSpan PolicyCheckFrequency = TimeSpan.FromSeconds(60);
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Operations/TimeSeries/TimeSeriesConfiguration.cs]: Change to property

File: src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs [hunk 2]
```diff
@@ -1620,6 +1620,78 @@ public void AddTimeSeriesNameToMetadata(DocumentsOperationContext ctx, string do
             }
         }
 
+        private static string EnsureOriginalName(DocumentsOperationContext ctx, string docId, string tsName)
+        {
+            var parts = tsName.Split(TimeSeriesConfiguration.TimeSeriesRollupSeparator);
```
Comment (ppekrol) [inline src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs:1718]: Can we have '@' in the name of TS?

Comment (karmeli87) [reply src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs:1718]: https://issues.hibernatingrhinos.com/issue/RavenDB-14800

**Review by ppekrol — APPROVED** [review]

**Comment by ppekrol** [comment]
I've rebased

**Review by ayende — APPROVED** [review]

File: src/Raven.Server/Documents/TimeSeries/TimeSeriesStats.cs [hunk 3]
```diff
@@ -70,7 +70,10 @@ public TimeSeriesStats(Transaction tx)
         private Table GetOrCreateTable(Transaction tx, CollectionName collection)
         {
             var tableName = collection.GetTableName(CollectionTableType.TimeSeriesStats); // TODO: cache the collection and pass Slice
-            TimeSeriesStatsSchema.Create(tx, tableName, 16);
+            
+            if (tx.IsWriteTransaction)
```
Comment (ppekrol) [inline src/Raven.Server/Documents/TimeSeries/TimeSeriesStats.cs:74]: I do not like this. Better to have 2 separate methods. Can't we have 2 separate methods? GetTable and GetOrCreateTable?

**Pull request merged**